### ISSUE TITLE
`unnecessary_cast`: `core::ffi::c_*` cases

### ISSUE THREAD

**Issue by ojeda** [rust-lang/rust-clippy#13494]
### Summary

`unnecessary_cast`'s does not cover some cases around `core::ffi::c_*` types that it could.

### Lint Name

`unnecessary_cast`

### Reproducer

I tried this code:

```rust
let x = 42;
let _ = x as core::ffi::c_int as core::ffi::c_int;
```

I expected to see this happen: lint triggers (e.g. "warning: casting to the same type is unnecessary (`core::ffi::c_int` -> `core::ffi::c_int`)").

Instead, this happened: lint does not trigger.

### Version

rustc 1.83.0-nightly (18b1161ec 2024-10-02)
binary: rustc
commit-hash: 18b1161ec9eeab8927f91405bca0ddf59a4a26c9
commit-date: 2024-10-02
host: x86_64-unknown-linux-gnu
release: 1.83.0-nightly
LLVM version: 19.1.0

**Comment by xtqqczze** [comment]
Confirmed as a false negative for `unnecessary_cast`.

The `core::ffi::c_*` case from the reproducer is still not linted by `clippy 0.1.100 (feaadeeaca 2026-09-19)`.

Reproducer: <https://rust.godbolt.org/z/PdrGKKW7G>
